### PR TITLE
Creating a pg_service.conf

### PR DESCRIPTION
**Pull request by SeqLaz**
This documentation is due to we receive many tickets requesting this steps

### CODE REVIEW

**Review by suricactus — COMMENTED** [review]
Did not check the contents of the PR yet, but what is the relation to the proposed documentation to https://docs.qfield.org/reference/qfieldcloud/secrets/?h=pg_service#pg_service-configuration ?

**Comment by SeqLaz** [comment]
@suricactus This is for creating the pg_service.conf 

> Did not check the contents of the PR yet, but what is the relation to the proposed documentation to https://docs.qfield.org/reference/qfieldcloud/secrets/?h=pg_service#pg_service-configuration ?

@suricactus this is the steps for creating an configuring the pg_service file in the computer and connecting to QGIS, this steps was requested many times in tickets, and this way also do reference to the secrets for using the same parameters to secrets in QFieldCloud

**Comment by lucienicolier** [comment]
@suricactus it turned out that many people were asking for support to sync a project with a postgis database on QFC, that's why I suggest @SeqLaz to add a step by step.

File: documentation/how-to/pg-service.en.md [hunk 1]
```diff
@@ -23,3 +23,73 @@ Usually the path on Android devices looks something like this: `/Android/data/ch
 QFieldCloud support `pg_service.conf` configurations too. You need to configure your PostgreSQL layers with "Offline editing" cloud action and store your service settings on QFieldCloud Project's Secrets page.
 
 Read more [how to configure PostgreSQL service](../reference/qfieldcloud/secrets.md) in the QFieldCloud documentation.
+
+## Creating a `pg_service.conf` File for PostgreSQL Connection in QGIS and Secrets
+
+Before beginning, ensure that you have allowed connections to QFieldCloud. Refer to [Technical specs](../reference/qfieldcloud/specs.md) for instructions.
+
+1. **Create a Text File**: Open your preferred text editor and create a new text file to define PostgreSQL service connection parameters. Save this file in your user's home directory, typically `~/.pg_service.conf`.
```
Comment (m-kuhn) [inline documentation/how-to/pg-service.en.md:31]: "home directory" and "~" are unix centric. Can we add instructions for windows users?

File: documentation/how-to/pg-service.en.md [hunk 2]
```diff
@@ -23,3 +23,73 @@ Usually the path on Android devices looks something like this: `/Android/data/ch
...
+   dbname=my_database
+   user=my_username
+   password=my_password
+   ```
+
+!![Parameters](../assets/images/service_config_file_001.png)
+
+3. **Save the File**: Save the file after adding connection parameters.
+
+4. **Environment Variable**: To ensure QGIS recognizes `pg_service.conf`, set the `PGSERVICEFILE` environment variable to point to its location. Add the following line to your shell's startup file (e.g., `~/.bashrc`, `~/.bash_profile`, `~/.zshrc`, etc.):
```
Comment (m-kuhn) [inline documentation/how-to/pg-service.en.md:63]: The environment variable is only required on windows, if we stick to the (unixish) example from above where a `.pg_service.conf` file is stored in the user folder and also what the shell's startup files involves, we don't need it.

Recommendation:
Split the docs where needed, this is partially done already (for windows), but the part here does not clearly indicated for who it is (and I think it should be much shorter).

Example:

### Setup pg_service.conf file

We first need to setup a configuration file. There are a lot of option to organize this, [read more in the PostgreSQL documentation](https://www.postgresql.org/docs/current/libpq-pgservice.html) or follow the description below.

**On windows**:
Create a file pg_service.conf and store it where you like

**On linux/macOS/*nix**

Create a file in `.pg_service.conf` in your home folder.

### Windows specific additional configuration steps

1. Create the environment variable to point to that (explanation, screenshot)
2. Logout/login

File: documentation/how-to/pg-service.en.md (hunk 1 above)
Comment (SeqLaz) [reply documentation/how-to/pg-service.en.md:31]: Yes, is already added the steps for the Windows users

File: documentation/how-to/pg-service.en.md (hunk 2 above)
Comment (SeqLaz) [reply documentation/how-to/pg-service.en.md]: That's greate, please review the new improvements

File: documentation/how-to/pg-service.en.md [hunk 3]
```diff
@@ -23,3 +23,76 @@ Usually the path on Android devices looks something like this: `/Android/data/ch
...
+   ```
+   [NINJA_DB]
+   host=my_db_for_ninja.ch
+   port=5432
+   dbname=my_database
+   user=my_username
+   password=my_password
+   ```
+
+!![Parameters](../assets/images/service_config_file_001.png)
```
Comment (suricactus) [inline documentation/how-to/pg-service.en.md:70]: I don't think this is really needed to be repeated twice, as well the blurred screenshot. Isn't it better if we remove 3.?

Comment (SeqLaz) [reply documentation/how-to/pg-service.en.md:70]: I completely agree!

Comment (SeqLaz) [reply documentation/how-to/pg-service.en.md]: I take new screeshots that looks a little bit better

**Comment by SeqLaz** [comment]
After review the checks that are not passing are links to Stackoverflow that require some captchas to enter, this links are in the [Specs](https://docs.qfield.org/reference/qfieldcloud/specs/) documentations

File: documentation/how-to/pg-service.en.md [hunk 4]
```diff
@@ -23,3 +23,64 @@ Usually the path on Android devices looks something like this: `/Android/data/ch
...
 
 Read more [how to configure PostgreSQL service](../reference/qfieldcloud/secrets.md) in the QFieldCloud documentation.
+
+## Creating a `pg_service.conf` File for PostgreSQL Connection in QGIS and Secrets
+
+Before beginning, ensure that your PostgreSQL database allows connections from QFieldCloud. Refer to [Technical specs](../reference/qfieldcloud/specs.md) for instructions.
+
+### Setup `pg_service.conf` File
+
+We first need to setup a configuration file. There are a lot of option to organize this, [read more in the PostgreSQL documentation](https://www.postgresql.org/docs/current/libpq-pgservice.html) or follow the description below.
```
Comment (lucienicolier) [inline documentation/how-to/pg-service.en.md:33]:
```suggestion
We first need to set up a configuration file. There are many options to organize this, [read more in the PostgreSQL documentation](https://www.postgresql.org/docs/current/libpq-pgservice.html) or follow the description below.
```

File: documentation/how-to/pg-service.en.md [hunk 5]
```diff
@@ -23,3 +23,64 @@ Usually the path on Android devices looks something like this: `/Android/data/ch
...
+   Replace placeholders (`your_host_or_ip`, `your_port`, `your_database_name`, `your_username`, `your_password`) with actual connection details and save the file.
+
+!![Parameters](../assets/images/service_config_file_001.png)
+
+### Additional Configuration Steps for Windows
+
+1. **Set Environment Variable**:
+   To ensure QGIS recognizes `pg_service.conf`, create an environment variable pointing to its location:
+
+   - Navigate to "This PC" or "My Computer" > Properties > Advanced system settings > Environment Variables.
```
Comment (lucienicolier) [inline documentation/how-to/pg-service.en.md:65]:
```suggestion
   - Navigate to "This PC" or "My Computer" > Properties > Advanced System Settings > Environment Variables.
```

**Review by suricactus — APPROVED** [review]
Looks good to me, better merge it and further improve it later.

**Pull request merged**